cmd/libs/go2idl/namer/plural_namer.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package namer
+
+import "k8s.io/kubernetes/cmd/libs/go2idl/types"
+
+type pluralNamer struct {
+	exceptions map[string]string
+	finalize   func(string) string
+}
+
+// NewPluralNamer returns a namer that returns the plural form of the input
+// type's name.
+func NewPluralNamer(exceptions map[string]string, public bool) *pluralNamer {
```

Comment:
To be honest, I vote for creating two separate New functions:

func NewPublicPluralNamer(exceptions)
func NewPrivatePluralNamer(exceptions)

having bool in such methods is always a bit more confusing.

WDYT?